registry/endpoint.go

```diff
@@ -227,6 +227,21 @@ func (e *Endpoint) pingV2() (RegistryInfo, error) {
 	}
 	defer resp.Body.Close()
 
+	// The endpoint may have multiple supported versions.
+	// Ensure it supports the v2 Registry API.
+	var supportsV2 bool
+
+	for _, versionName := range resp.Header[http.CanonicalHeaderKey("Docker-Distribution-API-Version")] {
```

Comment:
Yes, I think it's safe (accessing an unknown key from a map returns the default value for the mapped type, and it's safe to `range` over a `nil` slice).